app/services/paypal_service/store/paypal_account.rb

```diff
@@ -0,0 +1,220 @@
+module PaypalService::Store::PaypalAccount
+  PaypalAccountModel = ::PaypalAccount
+  OrderPermissionModel = ::OrderPermission
+
+  PaypalAccountCreate = EntityUtils.define_builder(
+    # Mandatory
+    [:community_id, :mandatory, :fixnum],
+    [:person_id, :optional, :string],
+    [:order_permission_request_token, :mandatory, :string],
+    [:order_permission_paypal_username_to, :mandatory, :string],
+
+    # Optional
+    [:active, one_of: [true, false, nil]],
+    [:email, :string],
+    [:payer_id, :string],
+
+    [:order_permission_verification_code, :string],
+    [:order_permission_scope, :string],
+
+    [:billing_agreement_billing_agreement_id, :string],
+    [:billing_agreement_paypal_username_to, :string],
+    [:billing_agreement_request_token, :string]
+  )
+
+  PaypalAccountUpdate = EntityUtils.define_builder(
+    [:active, :to_bool],
+    [:email, :string],
+    [:payer_id, :string],
+    [:order_permission_verification_code, :string],
+    [:order_permission_scope, :string],
+    [:billing_agreement_billing_agreement_id, :string],
+    [:billing_agreement_paypal_username_to, :string],
+    [:billing_agreement_request_token, :string]
+  )
+
+  PaypalAccount = EntityUtils.define_builder(
+    [:active, :mandatory, :to_bool],
+    [:community_id, :fixnum],
+    [:person_id, :string],
+    [:email, :string],
+    [:payer_id, :string],
+    [:order_permission_state, one_of: [:not_verified, :pending, :verified]],
+    [:billing_agreement_state, one_of: [:not_verified, :pending, :verified]],
+    [:billing_agreement_billing_agreement_id, :string]
+  )
+
+  # Rename map for PaypalAccount values that are stored to order_permissions table
+  ORDER_PERMISSIONS_MAP = {
+    order_permission_request_token: :request_token,
+    order_permission_paypal_username_to: :paypal_username_to,
+    order_permission_verification_code: :verification_code,
+    order_permission_scope: :scope
+  }
+
+  # Rename map for PaypalAccount values that are stored to billing_agreement table
+  BILLING_AGREEMENT_MAP = {
+    billing_agreement_billing_agreement_id: :billing_agreement_id,
+    billing_agreement_paypal_username_to: :paypal_username_to,
+    billing_agreement_request_token: :request_token
+  }
+
+  module_function
+
+  def create(opts:)
```

Comment:
The :opts keyword param feels a bit unnecessary since it's the only parameter and also generic name opts doesn't add anything to document the call site.